test/Image_Tests/src/Data/Image_Spec.enso

```diff
@@ -0,0 +1,232 @@
+            eye.get 1 0 . should_equal [0]
+            eye.get 1 1 . should_equal [1]
+            Test.expect_panic_with (eye.get 10 10) Matrix.Index_Out_Of_Bounds_Error
+            Test.expect_panic_with (eye.get -1 -1) Matrix.Index_Out_Of_Bounds_Error
+
+        Test.specify "should add scalar" <|
+            zeros+1 . should_equal ones
+            ones+1 . should_equal ones
+            eye+1 . should_equal ones
+            eye+0 . should_equal eye
+            zeros+0.8 . should_equal (Image.from_vector (Vector.fill 9 0.8) channels=1 rows=3)
+        Test.specify "should add vector" <|
+            zeros+[1] . should_equal ones
+            ones+[1, 1] . should_equal ones
+            eye+[0, 0] . should_equal eye
+            eye+[] . should_equal eye
+
+        Test.specify "should subtract scalar" <|
+            ones-1 . should_equal zeros
+            zeros-1 . should_equal zeros
+            ones-1 . should_equal zeros
+        Test.specify "should subtract vector" <|
+            zeros-[1] . should_equal zeros
+            ones-[1, 1] . should_equal zeros
+            eye-[0, 0] . should_equal eye
+            eye-[] . should_equal eye
+
+        Test.specify "should multiply scalar" <|
+            ones*2 . should_equal ones
+            zeros*2 . should_equal zeros
+            eye*0 . should_equal zeros
+            eye*1 . should_equal eye
+        Test.specify "should multiply vector" <|
+            zeros*[2] . should_equal zeros
+            ones*[1, 1] . should_equal ones
+            eye*[0, 0] . should_equal zeros
+            eye*[] . should_equal zeros
+
+        Test.specify "should divide scalar" <|
+            zeros/2 . should_equal zeros
+            ones/5 . should_equal (Image.from_vector (Vector.fill 9 0.2) channels=1 rows=3)
+        Test.specify "should divide vector" <|
+            zeros/[2] . should_equal zeros
+            ones/[5, 5] . should_equal (Image.from_vector (Vector.fill 9 0.2) channels=1 rows=3)
+            eye/[] . should_equal zeros
+
+    Test.group "Image channels 2" <|
+        zeros = Image.from_vector (Vector.fill 9 0 . flat_map x->[x,x]) rows=3 channels=2
+        ones = Image.from_vector (Vector.fill 9 1 . flat_map x->[x,x]) rows=3 channels=2
+        eye = Image.from_vector ([1, 0, 0, 0, 1, 0, 0, 0, 1] . flat_map x->[x,0]) rows=3 channels=2
+        Test.specify "should convert to vector" <|
+            zeros.to_vector . should_equal (Vector.fill 9*2 0)
+            ones.to_vector . should_equal (Vector.fill 9*2 1)
+            eye.to_vector . should_equal [1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 1, 0]
+        Test.specify "should get value" <|
+            eye.get 0 0 . should_equal [1, 0]
+            eye.get 1 0 . should_equal [0, 0]
+            eye.get 1 1 . should_equal [1, 0]
+            Test.expect_panic_with (eye.get 10 10) Matrix.Index_Out_Of_Bounds_Error
+            Test.expect_panic_with (eye.get -1 -1) Matrix.Index_Out_Of_Bounds_Error
```

Comment:
I don't think that this should result in a panic. I would do this as a dataflow error as users may run into this.